src/System.Private.CoreLib/shared/System/Runtime/AmbiguousImplementationException.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime;
+
+namespace System.Runtime.Serialization
+{
+    [Serializable]
+    public sealed class AmbiguousImplementationException : Exception
+    {
+        private static string s_nullMessage = SR.AmbiguousImplementationException_NullMessage;
```

Comment:
This is the behavior for `SerializationException`, which I based this class on: https://github.com/dotnet/coreclr/blob/69c8421c1177675d4df855d7bb73b0a77b19c643/src/System.Private.CoreLib/shared/System/Runtime/Serialization/SerializationException.cs#L13

Should I just directly use SR.Ambiguous... instead?

Reply:
Yes, I think so. 

It would be fine to undo the caching for serialization exception as well. It is an outlier and it does not provide any real benefits.